hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreEngine.java

```diff
@@ -514,4 +514,12 @@ public void removeCompactedFiles(Collection<HStoreFile> compactedFiles) {
       throw new IOException("Unable to load configured store engine '" + className + "'", e);
     }
   }
+
+  public boolean requireWritingToTmpDirFirst() {
+    return storeFileTracker.requireWritingToTmpDirFirst();
+  }
+
+  public void resetCompactionWriter(){
```

Comment:
And a javadoc comment here as this is critical to ensuring that we know which files are sane to be deleted, please. Touch on `doCompaction` calling this.